test/models/test_step.py

```diff
@@ -73,49 +82,54 @@ def test_wrong_material_matchrules(self):
     self.assertFalse(self.step.validate())
 
   def test_wrong_product_matchrules(self):
-    """ test that the product matchrule validatores catch malformed ones """
+    """Test that the product matchrule validators catch malformed ones."""
 
+    self.step.product_matchrules = [["NONFOO"]]
     with self.assertRaises(FormatError):
-      self.step.product_matchrules = [["NONFOO"]]
       self.step._validate_product_matchrules()
+
+    with self.assertRaises(FormatError):
       self.assertFalse(self.step.validate())
 
-      self.step.product_matchrules = "PFF"
+    self.step.product_matchrules = "PFF"
+    with self.assertRaises(FormatError):
       self.step._validate_product_matchrules()
+
+    with self.assertRaises(FormatError):
       self.assertFalse(self.step.validate())
 
     # for more thorough tests, check the test_matchrule.py module
     self.step.product_matchrules = [["CREATE", "foo"]]
     self.step._validate_product_matchrules()
     self.assertFalse(self.step.validate())
```

Comment:
I wonder why the test does not fail here. Shouldn't validate always return `None`?